### ISSUE TITLE
Restart functionality for certificate renewal

### ISSUE THREAD

**Issue by alinadonisa** [cockroachdb/cockroach-operator#367]
1. Rolling restart
2. Full restart logic for CA

**Comment by alinadonisa** [comment]
@keith-mcclellan please add more details on the logic of the restart

**Comment by keith-mcclellan** [comment]
This is correct, the cluster restart api should have two types of restarts - rolling (roll pods while respecting the podDisruptionBudget) and full (restart all pods irrespective of the podDisruptionBudget)
